pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -558,10 +582,49 @@ func getDashbuilderTemplate(cr *api.KieApp, serversConfig []api.ServerTemplate,
 			dashbuilderTemplate.Cors = *cr.Status.Applied.Objects.Dashbuilder.Cors
 		}
 
+		// Enable EdgeTermination
+		if cr.Status.Applied.Objects.Dashbuilder.TerminationRoute != nil && len(cr.Status.Applied.Objects.Dashbuilder.TerminationRoute.Protocol) > 0 {
+			dashbuilderTemplate.TerminationRoute.Termination, dashbuilderTemplate.TerminationRoute.Protocol = getDashbuilderTermination(cr)
+			dashbuilderTemplate.TerminationRoute.EnableTlsSection = true
+			if len(cr.Status.Applied.Objects.Dashbuilder.TerminationRoute.Key) > 0 {
+				dashbuilderTemplate.TerminationRoute.Key = cr.Status.Applied.Objects.Dashbuilder.TerminationRoute.Key
+			}
+			if len(cr.Status.Applied.Objects.Dashbuilder.TerminationRoute.Certificate) > 0 {
+				dashbuilderTemplate.TerminationRoute.Certificate = cr.Status.Applied.Objects.Dashbuilder.TerminationRoute.Certificate
+			}
+			if len(cr.Status.Applied.Objects.Dashbuilder.TerminationRoute.CaCertificate) > 0 {
+				dashbuilderTemplate.TerminationRoute.CaCertificate = cr.Status.Applied.Objects.Dashbuilder.TerminationRoute.CaCertificate
+			}
+
+		} else {
+			// legacy behaviour
+			dashbuilderTemplate.TerminationRoute.Termination, dashbuilderTemplate.TerminationRoute.Protocol = getDashbuilderTermination(cr)
+			if cr.Status.Applied.CommonConfig.DisableSsl {
+				dashbuilderTemplate.TerminationRoute.EnableTlsSection = false
+			} else {
+				dashbuilderTemplate.TerminationRoute.EnableTlsSection = true
+			}
+		}
+
 	}
 	return dashbuilderTemplate, nil
 }
 
+func getDashbuilderTermination(cr *api.KieApp) (string, string) {
+	if cr.Status.Applied.Objects.Dashbuilder.TerminationRoute != nil {
+		if len(cr.Status.Applied.Objects.Dashbuilder.TerminationRoute.Protocol) > 0 {
+			// Edge is always http
+			return constants.EdgeTermination, constants.HttpProtocol
```

Comment:
it does not sounds right to me, why always return edge and http?
if user just wants a default http route?

same applies for all other similar methods.

Reply:
Ok, better to clarify using a boolean with enableEdge, this code was modeled reading the protocol http or https